presto-orc/src/test/java/io/prestosql/orc/BenchmarkStreamReaders.java

```diff
@@ -1096,6 +1458,43 @@ private OrcRecordReader createRecordReader()
         }
     }
 
+    @State(Scope.Thread)
+    public static class TimestampWithAllNullBenchmarkData
+    {
+        protected File temporaryDirectory;
+        protected File timestampWithAllNullFile;
+
+        @Setup
+        public void setup()
+                throws Exception
+        {
+            temporaryDirectory = createTempDir();
+
+            timestampWithAllNullFile = new File(temporaryDirectory, randomUUID().toString());
+            writeOrcColumnHive(timestampWithAllNullFile, ORC_12, NONE, TIMESTAMP, NULL_VALUES.iterator());
+        }
+
+        @TearDown
+        public void tearDown()
+                throws IOException
+        {
+            deleteRecursively(temporaryDirectory.toPath(), ALLOW_INSECURE);
+        }
+
+        private OrcRecordReader createRecordReader()
+                throws IOException
+        {
+            OrcDataSource dataSource = new FileOrcDataSource(timestampWithAllNullFile, new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE), true);
```

Comment:
Is it possible to wrap this into some static method so that there is less repetition in the benchmark?

Reply:
This can be done I'll update it

Reply:
It might be possible to use the same base class for this, and only swap out the `Type` parameter.